### PR TITLE
Fixes #6030 - cloner autocloning people who have ghosted/cryoed

### LINKED ISSUE #6030: Cloners can produce braindead clones of SSD players
**Problem Description**:
If you cryo then ghost, after being scanned by an autoprocess cloner, it can create a braindead clone of you against your wishes.

**What did you expect to happen**: 
Upon cryoing and ghosting out (long term storage), the character's record should be deleted from all cloning consoles that have it.

**What happened instead**:
They kept the record, and immediately cloned the character.

**Why is this bad/What are the consequences:**
Characters can be cloned against their players' wishes, sometimes even without their knowledge. This results in braindead clones.

**Steps to reproduce the problem**:
1. Get scanned in an autoprocess cloner
2. Go cryo and ghost out.
3. Wait a few minutes - a clone will be created for you even though you deliberately left the round

### PR DESCRIPTION
**Pull request by Kyep**
Currently, if you are scanned by an autoprocess cloner, then go to cryo and ghost out, late in a round, then the cloner can still autoclone you.

Obviously, this is not desirable - you deliberately left the round for a reason.

This PR fixes the "ghost" verb such that ghosting despite the "you can't re-enter the round after this" warning, whether in cryo or not, prevents the cloner from autocloning you.

🆑 Kyep
fix: Autoprocess cloners will no longer clone people who cryo and ghost out, or otherwise deliberately remove themselves from the round.
/🆑

### CODE REVIEW

**Comment by Krausus** [comment]
If you get the warning before ghosting, you're flagged as non-respawnable, which the cloner already checks for; removing the records after the warning doesn't change anything.

On the other hand, removing the records in the cryopod's `despawn_occupant` proc should work, though you may need to grab the ckey from the body's ghost.

**Comment by Kyep** [comment]
@Krausus Why even keep the record in the cloner if they're non-respawnable anyway? Its just another thing to loop through every time. A non-respawnable ghost shouldn't be able to become a respawnable one.

I put the code in verb/ghost rather than cryopod, because I worried that if I only put it in the cryopod, people could get prescanned, ghost out of their body, hear deadchat / see everything, and then be auto-cloned. This would effectively function as an exploit allowing people to get info they couldn't otherwise.

**Comment by tigercat2000** [comment]
Because DNA records shouldn't just randomly disappear- It'll be confusing to medical players.

**Comment by Krausus** [comment]
> Why even keep the record in the cloner if they're non-respawnable anyway?

Because, like Tiger said, the simple act of irreversibly ghosting should not have unexplained side effects elsewhere on the station. Removing the records when someone is despawned by cryo is fine because that has an explicit notification when it removes all traces of them.

File: code/game/machinery/computer/cloning.dm [hunk 1]
```diff
@@ -1,3 +1,5 @@
+var/list/clonecomputers = list()
```
Comment (Fox-McCloud) [inline code/game/machinery/computer/cloning.dm]: This isn't really needed; the `machines` list is small enough to search through that we really don't need a specific list for clone computers.

File: code/game/machinery/cryopod.dm [hunk 2]
```diff
@@ -290,6 +290,11 @@
 					continue
 				O.loc = src
 
+	for(var/obj/machinery/computer/cloning/cloner in clonecomputers)
+		for(var/datum/dna2/record/R in cloner.records)
+			if(R.ckey == occupant.ckey)
```
Comment (Krausus) [inline code/game/machinery/cryopod.dm]: If someone ghosts in cryo, they'll be out of their body before this, and `occupant.ckey` will be null. Comparing `mind`s might do the trick, though.

**Comment by Fox-McCloud** [comment]
Going to address the concerns here?

**Comment by Kyep** [comment]
@Fox-McCloud @Krausus Take a look.

File: code/game/machinery/cryopod.dm [hunk 3]
```diff
@@ -290,6 +290,11 @@
 					continue
 				O.loc = src
 
+	for(var/obj/machinery/computer/cloning/cloner in machines)
+		for(var/datum/dna2/record/R in cloner.records)
+			if(occupant.mind == locate(R.mind))
```
Comment (Crazylemon64) [inline code/game/machinery/cryopod.dm]: Wait a minute this should just be `== R.mind`

I'm not sure how `locate` would behave in a situation like this - [the doc only talks about the cases of either text or path arguments, and nothing of an instance argument.](http://www.byond.com/docs/ref/info.html#/proc/locate)

Comment (Crazylemon64) [inline code/game/machinery/cryopod.dm]: [Darn it cloning code, stop being weird](https://github.com/ParadiseSS13/Paradise/blob/master/code/game/machinery/computer/cloning.dm#L424-L425)

**Pull request merged**